app/src/components/Layout/Box.js

```diff
@@ -0,0 +1,77 @@
+import { useState, Fragment, useRef } from "react";
+
+import Form from "../Forms/Form";
+import Button from "../UI/Button";
+import { fetchSearches } from "../../createExcelFiles/fetchSearches";
+import "./Box.css";
+
+function Box() {
+  const queryInputRef = useRef();
+  const searchNumberInputRef = useRef();
+  const [showForm, setShowForm] = useState(true);
+  // const [isLoading, setIsLoading] = useState(false);
+
+  const onSubmitHandler = (event) => {
+    event.preventDefault();
+
+    if (
+      queryInputRef.current.value.length > 0 &&
+      searchNumberInputRef.current.value >= 1 &&
+      searchNumberInputRef.current.value <= 99
+    ) {
+      fetchSearches(queryInputRef.current.value,searchNumberInputRef.current.value);
```

Comment:
So I understand this correctly. You are fetching the results from an API and then downloading it through your excel library. The biggest issue here is your naming. You should keep functions small and to the point. You are "fetching" data in your `fetch` function but you are also processing and downloading it. Its doing too many things the `fetchSearches` function

```javascript
if(
 queryInputRef.current.value.length > 0 &&
 searchNumberInputRef.current.value >= 1 &&
 searchNumberInputRef.current.value <= 99) {
 try  {
  const googleResult = await fetchSearches(queryInputRef.current.value,searchNumberInputRef.current.value);
  await downloadFile(googleResult); //Create another function that just calls the XSLX to download the file
  } catch {
   setError(true); // I would also handle error here in the UI to let the user know if there is any issue
 }
}
```